### DISCUSSION TITLE
Extension Activation Failed on Remote EC2: `getaddrinfo EAI_AGAIN api.github.com`

### DISCUSSION THREAD

**Discussion by panizbehboudian** [community/community#69820] [Copilot Conversations]
### Select Topic Area

Bug

### Body

**Issue:**

I've run into an issue with GitHub Copilot activation when using it on a remote EC2 instance. The error suggests there's a DNS resolution problem when the extension tries to fetch a token for Copilot.

**Environment:**
- Host: AWS EC2 instance.
- VS Code version: 1.83.0
- GitHub Copilot extension version: 1.125.0 (as per logs).

**Description:**

After updating VS Code on my EC2 instance, GitHub Copilot failed to activate. This issue wasn't present on my local machine. The error logs provide the following detailed trace:

```
FetchError: getaddrinfo EAI_AGAIN api.github.com
    at fetch (/home/ubuntu/.vscode-server/extensions/github.copilot-1.125.0/node_modules/@adobe/helix-fetch/src/fetch/index.js:99:11)
    ...
    at s.value (/home/ubuntu/.vscode-server/extensions/github.copilot-1.125.0/extension/src/extension.ts:169:9)
```

The error seems to arise when the extension is trying to fetch a Copilot token using the `@adobe/helix-fetch` library.

**Steps to Reproduce:**
1. Connect to the EC2 instance.
2. Launch VS Code with the GitHub Copilot extension installed.
3. Observe the error upon activation attempt.

**Expected Behavior:**

GitHub Copilot should activate seamlessly and function as expected.

**Actual Behavior:**

I'm presented with a `getaddrinfo EAI_AGAIN api.github.com` error, which seems to be related to DNS resolution during the token fetching process.

**Additional Information:**

- This issue started after updating VS Code on the EC2 instance.
- Copilot works flawlessly on my local machine.
- I have already tried re-installing the extension and also signing out and in again.
Would appreciate any assistance or insights regarding this problem. Thank you!

**Comment by Uliana200407** [comment]
Hello, so that can be a trouble with your Network, check the network configuration on your EC2 instance. Also ensure it resolve DNS addresses.You can test it with ping on api.github.com.If you will get no reply - that is the trouble with your network
Also check the proxy, they should be configured correctly in EC2 instance.
So, maybe it will help you

**Reply by panizbehboudian** [reply]
The irony is that I haven't changed my instance network configuration and copilot was working without any problems until I updated my VSCode.

**Comment by panizbehboudian** [comment]
The problem resolved with changing to pre-release version. But then again stopped working today!
